parser-typechecker/src/Unison/Codebase/SqliteCodebase.hs

```diff
@@ -736,18 +770,15 @@ sqliteCodebase debugName root = do
             pure $ Set.map (Causal.RawHash . Cv.hash2to1 . unCausalHash) cs
 
           sqlLca :: MonadIO m => Branch.Hash -> Branch.Hash -> m (Maybe Branch.Hash)
-          sqlLca h1 h2 = liftIO $ Control.Exception.bracket open close \(c1, c2) ->
-            runDB conn
-              . (fmap . fmap) Cv.causalHash2to1
-              $ Ops.lca (Cv.causalHash1to2 h1) (Cv.causalHash1to2 h2) c1 c2
-            where
-              open = (,) <$> unsafeGetConnection (debugName ++ ".lca.left") root
-                         <*> unsafeGetConnection (debugName ++ ".lca.left") root
-              close (c1, c2) = shutdownConnection c1 *> shutdownConnection c2
-
+          sqlLca h1 h2 =
+            liftIO $ withConnection  (debugName ++ ".lca.left") root $ \c1 -> do
```

Comment:
This also fixes a (small) bug: when acquiring two resources, you don't want to bundle the acquire/release actions, because a failure to acquire the second action will cause the first to go unreleased.